File: Services/Collector/Models/Dto/EventDto.cs
namespace Wingrid.Services.Collector.Models.Dto
{
    public class EventDto
    {
        // Event
        public string? Id { get; set; }
        public DateTime? Date { get; set; }
        public string? Name { get; set; }
        public string? ShortName { get; set; }

        // Season
        public int? SeasonType { get; set; }
        public int? Season { get; set; }
        public string? SeasonSlug { get; set; }

        // Week
        public int? Week { get; set; }

        // Competition
        public int? Attendance { get; set; }
        public bool? NeutralSite { get; set; }
        public bool? TimeValid { get; set; }
        public bool? ConferenceCompetition { get; set; }
        public bool? PlayByPlayAvailable { get; set; }
        public bool? Recent { get; set; }

        // Home Competitor
        public bool? HomeWinner { get; set; }
        public string? HomeTeamId {get; set;}
        public Team? HomeTeam { get; set; }
        public string? HomeScore { get; set; }

        // Away Competitor
        public bool? AwayWinner { get; set; }
        public string? AwayTeamId {get; set;}
        public Team? AwayTeam { get; set; }
        public string? AwayScore { get; set; }

        // Status
        // public int? Clock { get; set; }
        public string? DisplayClock { get; set; }
        public int? Period { get; set; }

        // Status Type
        public string? StatusId { get; set; }
        public string? StatusName { get; set; }
        public string? StatusState { get; set; }
        public bool? StatusCompleted { get; set; }
        public string? StatusDescription { get; set; }
        public string? StatusDetail { get; set; }
        public string? StatusShortDetail { get; set; }
    }
}

File: Source/Web/ClientApp/src/models/EventDto.ts
import { TeamDto } from "./TeamDto";

export interface EventDto {
    // Event
    id: string;
    date: string;
    name: string;
    shortName: string;

    // Season
    seasonType: number;
    season: number;
    seasonSlug: string;

    // Week
    week: number;

    // Competition
    attendance: number;
    neutralSite: boolean;
    timeValid: boolean;
    conferenceCompetition: boolean;
    playByPlayAvailable: boolean;
    recent: boolean;

    // Home Competitor
    homeWinner: boolean;
    homeTeamId: string;
    homeTeam: TeamDto;
    homeScore: string;

    // Away Competitor
    awayWinner: boolean;
    awayTeamId: string;
    awayTeam: TeamDto;
    awayScore: string;

    // Status
    displayClock: string;
    period: number;

    // Status Type
    statusId: string;
    statusName: string;
    statusState: string;
    statusCompleted: boolean;
    statusDescription: string;
    statusDetail: string;
    statusShortDetail: string;
}
